libraries/AP_Proximity/AP_Proximity_LD06.cpp

```diff
@@ -0,0 +1,183 @@
+
+                set_status(AP_Proximity::Status::Good);
+            }
+        } else {
+            gcs().send_text(MAV_SEVERITY_CRITICAL, "LD06 Invalid Packet: %d", c);
+        }
+    }
+
+    // Store for debugging purposes
+    _last_request_ms = AP_HAL::millis();
+}
+
+// Parses the data packet received from the LiDAR
+void AP_Proximity_LD06::parse_response_data()
+{ 
+
+    // Data interpretation based on:
+    // http://wiki.inno-maker.com/display/HOMEPAGE/LD06?preview=/6949506/6949511/LDROBOT_LD06_Development%20manual_v1.0_en.pdf
+
+    // Second byte in array stores length of data - not used but stored for debugging
+    // const uint8_t data_length = _response[_start_data_length];
+
+    // Respective bits store the radar speed, start/end angles
+    // Use bitwise operations to correctly obtain correct angles
+    // Divide angles by 100 as per manual
+    // const uint16_t radar_speed = UINT16_VALUE(_response[_start_radar_speed + 1], _response[_start_radar_speed]);
+    const float start_angle = float(UINT16_VALUE(_response[START_BEGIN_ANGLE + 1], _response[START_BEGIN_ANGLE]))  * 0.01;
+    const float end_angle = float(UINT16_VALUE(_response[START_END_ANGLE + 1], _response[START_END_ANGLE])) * 0.01;
+
+    // Verify the checksum that is stored in the last element of the response array
+    // Return if checksum is incorrect - i.e. bad data, bad readings, etc.
+    uint8_t check_sum = _response[START_CHECK_SUM];
+    if(check_sum != calc_checksum(&_response[0], sizeof(_response)/sizeof(_response[0]) - 1)) {
+        gcs().send_text(MAV_SEVERITY_CRITICAL, "LD06 Bad Check Sum: %d", check_sum);
+        return;
+    }
+
+    // Used to calculate the angle inside the data packet
+    uint8_t angle_inc_counter = 0;
+
+    // Each recording point is three bytes long, goes through all of that and updates database
+    for (uint16_t i = START_PAYLOAD; i < START_PAYLOAD + MEASUREMENT_PAYLOAD_LENGTH * PAYLOAD_COUNT; i += MEASUREMENT_PAYLOAD_LENGTH) {
+
+        // Gets the distance recorded and converts to meters
+        const uint16_t distance_mm = (_response[i + 1] << 0x08) | (_response[i]);
+        float distance = distance_mm * 0.001f;
+
+        // Calculates the angle that this point was sampled at
+        const float sampled_angle = start_angle + angle_inc_counter * (end_angle - start_angle) /  (PAYLOAD_COUNT - 1);
+        angle_inc_counter++;
+
+        // Validates data and checks if it should be included
+        if (distance > distance_min() && distance < distance_max()) {
+            if (ignore_reading(sampled_angle, distance)) {
+                gcs().send_text(MAV_SEVERITY_CRITICAL, "LD06 Ignored Reading- Angle: %.2f  Distance: %.2f", sampled_angle, distance);
+                continue;
+            }
+            // Convert angle to approrpriate face and adds to database
+            gcs().send_text(MAV_SEVERITY_CRITICAL, "LD06 Accepted Reading- Angle: %.2f  Distance: %.2f", sampled_angle, distance);
```

Comment:
We won't want this in the final code so add a note to remember to remove it ;)
Too much spam.  

```suggestion
//   REMOVE ME FROM FINAL CODE
            GCS_SEND_TEXT(MAV_SEVERITY_CRITICAL, "LD06 Accepted Reading- Angle: %.2f  Distance: %.2f", sampled_angle, distance);
```